File: project/frontend/sharespot-device-records-frontend/project/src/app/mappers/DeviceMapper.ts
import {Device} from "../model/Device";
import {DeviceIdDTO} from "../dtos/RecordsDTO";

export class DeviceMapper {

  static dtoToModel(dto: DeviceIdDTO): Device {
    return new Device(dto.deviceId);
  }

  static modelToDto(model: Device): DeviceIdDTO {
    return {deviceId: model.deviceId}
  }
}

File: project/backend-services/sharespot-identity-management-backend/application/src/main/java/sharespot/services/identitymanagementbackend/application/mapper/device/DeviceMapper.java
package sharespot.services.identitymanagementbackend.application.mapper.device;

import sharespot.services.identitymanagementbackend.application.model.device.DeviceIdDTO;
import sharespot.services.identitymanagementbackend.application.model.device.ExpelDeviceFromDomainDTO;
import sharespot.services.identitymanagementbackend.application.model.device.PlaceDeviceInDomainDTO;
import sharespot.services.identitymanagementbackend.domainservices.model.device.DeviceResult;
import sharespot.services.identitymanagementbackend.domainservices.model.device.PlaceDeviceInDomainCommand;
import sharespot.services.identitymanagementbackend.domainservices.model.device.RemoveDeviceFromDomainCommand;

public interface DeviceMapper {

    PlaceDeviceInDomainCommand dtoToCommand(PlaceDeviceInDomainDTO dto);
    
    RemoveDeviceFromDomainCommand dtoToCommand(ExpelDeviceFromDomainDTO dto);
    
    DeviceIdDTO resultToDto(DeviceResult result);
}
